File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/RsaGenerationParameters.java
package org.cloudfoundry.credhub.requests;

import java.util.Objects;

public class RsaGenerationParameters extends RsaSshGenerationParameters {
  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }

    if (o == null || getClass() != o.getClass()) {
      return false;
    }

    final RsaGenerationParameters that = (RsaGenerationParameters) o;
    return Objects.equals(getKeyLength(), that.getKeyLength());
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/RsaGenerationParameters.kt
package org.cloudfoundry.credhub.requests

import java.util.Objects

class RsaGenerationParameters : RsaSshGenerationParameters() {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }

        if (other == null || javaClass != other.javaClass) {
            return false
        }

        val that = other as RsaGenerationParameters?
        return keyLength == that!!.keyLength
    }

    override fun hashCode(): Int = Objects.hash(keyLength)
}
